# # Error Handling
### <p style="color:Tomato">Learned the basics of error handling while finding common names for U.S. Congress members.<p/>
some basic exploration and manipulation with legislators.csv, and laid the groundwork for our "names project".

In [1]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

#### <p style="color:Gray">Missing Data<p/><hr>
working with **legislators.csv** <br/>
:which records information on every historical member of the U.S. Conress.

some rows contain misiing values for some columns.<br/>

> we'll explore some of the errors that occur when we ignore missing values and how to handle them

When exploring data, it's often useful to extract the unique elements in a list.<br/>
데이터를 탐색 할 때 고유 한 요소를 목록에서 추출하는 것이 유용한 경우가 많습니다. 

#### <p style="color:Gray">Set<p/><hr>
If you add an element to a set that it already contains, the set will ignore it. <br/>
The items in a set are unordered, while each item in a list has an index.<br/>
Sets don't have indexes, the items in a set may display in a different order each time you print it.<br/>
<br/>
이미 포함 된 집합에 요소를 추가하면 집합에서이를 무시합니다.<br/>
집합의 항목은 정렬되지 않은 반면, 목록의 각 항목에는 색인이 있습니다. Sets에는 인덱스가 없으므로 세트를 인쇄 할 때마다 세트의 항목이 다른 순서로 표시 될 수 있습니다.

In [2]:
["Dog", "Cat", "Hippo", "Dog", "Cat", "Dog", "Dog", "Cat"]
unique_animals = set(["Dog", "Cat", "Hippo", "Dog", "Cat", "Dog", "Dog", "Cat"])
print(unique_animals)

['Dog', 'Cat', 'Hippo', 'Dog', 'Cat', 'Dog', 'Dog', 'Cat']

{'Dog', 'Hippo', 'Cat'}


In [3]:
unique_animals.add('Tiger')
unique_animals.remove('Dog')
print(unique_animals)
list(unique_animals)

{'Tiger', 'Hippo', 'Cat'}


['Tiger', 'Hippo', 'Cat']

In [4]:
import csv
f = open('legislators.csv', 'r')
csvreader = csv.reader(f)
legi_data = list(csvreader)
legi_data[:10]

[['last_name', 'first_name', 'birthday', 'gender', 'type', 'state', 'party'],
 ['Bassett', 'Richard', '1745-04-02', 'M', 'sen', 'DE', 'Anti-Administration'],
 ['Bland', 'Theodorick', '1742-03-21', '', 'rep', 'VA', ''],
 ['Burke', 'Aedanus', '1743-06-16', '', 'rep', 'SC', ''],
 ['Carroll', 'Daniel', '1730-07-22', 'M', 'rep', 'MD', ''],
 ['Clymer', 'George', '1739-03-16', 'M', 'rep', 'PA', ''],
 ['Contee', 'Benjamin', '', 'M', 'rep', 'MD', ''],
 ['Dalton', 'Tristram', '1738-05-28', '', 'sen', 'MA', 'Pro-Administration'],
 ['Elmer', 'Jonathan', '1745-11-29', 'M', 'sen', 'NJ', 'Pro-Administration'],
 ['Few', 'William', '1748-06-08', 'M', 'sen', 'GA', 'Anti-Administration']]

In [5]:
class Dataset:
    def __init__(self, data):
        self.header = data[0]
        self.data = data[1:]
    def column(self, label):
        if label not in self.header:
            return None
        index = 0
        for idx, element in enumerate(self.header):
            if label == element:
                index = idx
        column = []
        for row in self.data:
            column.append(row[index])
        return column
    def count_unique(self, label):
        unique_results = set(self.column(label))
        count = len(unique_results)
        return count
    def set_column(self, label):
        unique_results = set(self.column(label))
        return unique_results

In [6]:
legi_dataset = Dataset(legi_data)
header = legi_dataset.header
gender = legi_dataset.column('gender')
print(header)
print(gender[:10])

['last_name', 'first_name', 'birthday', 'gender', 'type', 'state', 'party']
['M', '', '', 'M', 'M', 'M', '', 'M', 'M', 'M']


In [7]:
gender = set(gender)
print(gender)

{'', 'M', 'F'}


In [8]:
gender_def = legi_dataset.set_column('gender')
print(gender_def)

{'', 'M', 'F'}


In [9]:
gender = []
for item in legi_data[1:]:
    gender.append(item[3])
    
gender = set(gender)
print(gender)

{'', 'M', 'F'}


<p style="color:Tomato">I need to remove null values<p/>
<p style="color:Gray">Print out datas and inspect the data<p/><hr>
working with **legislators.csv** <br/>
:which records information on every historical member of the U.S. Conress.

some rows contain misiing values for some columns.<br/>

> we'll explore some of the errors that occur when we ignore missing values and how to handle them
When you have a fresh dataset, it's always a good idea to look for any patterns.
* Missing data
    * some files contain empty fields. Others may use a string like N/A to indicate missing values.
* Values that don't make intuitive sense
    * A legislator with a birhday in 2050, for example, would indicate a problem with the data.
* Recurring themes 
    * One theme in this dataset is that the overwhelming majority of legislators are male.

<br/>

새로운 데이터 세트를 가지고있을 때는 항상 다음과 같은 패턴을 찾아 보는 것이 좋습니다.
* 누락된 데이터
    * 일부 파일에는 빈 필드가 있습니다. 다른 사람들은 N / A와 같은 문자열을 사용하여 누락 된 값을 나타낼 수 있습니다.
* 직관적으로 이해할 수 없는 가치
    * 예를 들어, 2050 년 생일을 가진 입법자는 데이터에 문제가 있음을 나타냅니다.
* 되풀이되는 테마
    * 이 데이터 세트의 하나의 주제는 압도적으로 다수의 의원이 남성이라는 점입니다. 

In [10]:
header = legi_dataset.header
print(header)

['last_name', 'first_name', 'birthday', 'gender', 'type', 'state', 'party']


In [11]:
legislators = legi_dataset.data
print(legislators[:10])

[['Bassett', 'Richard', '1745-04-02', 'M', 'sen', 'DE', 'Anti-Administration'], ['Bland', 'Theodorick', '1742-03-21', '', 'rep', 'VA', ''], ['Burke', 'Aedanus', '1743-06-16', '', 'rep', 'SC', ''], ['Carroll', 'Daniel', '1730-07-22', 'M', 'rep', 'MD', ''], ['Clymer', 'George', '1739-03-16', 'M', 'rep', 'PA', ''], ['Contee', 'Benjamin', '', 'M', 'rep', 'MD', ''], ['Dalton', 'Tristram', '1738-05-28', '', 'sen', 'MA', 'Pro-Administration'], ['Elmer', 'Jonathan', '1745-11-29', 'M', 'sen', 'NJ', 'Pro-Administration'], ['Few', 'William', '1748-06-08', 'M', 'sen', 'GA', 'Anti-Administration'], ['Floyd', 'William', '1734-12-17', 'M', 'rep', 'NY', '']]


In [12]:
# Print out party and inspect the values. And print out legislators and inspect the data.
party = []
for item in legislators:
    party.append(item[6])
party = set(party)    
print(party)
print(legislators[:10])


{'', 'Democrat', 'American Labor', 'Independent', 'Unconditional Unionist', 'Independent Democrat', 'Anti-Jacksonian', 'Anti Jacksonian', 'Adams', 'Anti-Administration', 'Democratic and Union Labor', 'Progressive Republican', 'Republican', 'Jacksonian', 'Ind. Whig', 'Jackson Republican', 'Union Democrat', 'Readjuster', 'National Greenbacker', 'Prohibitionist', 'Whig', 'Union', 'Conservative Republican', 'Anti Jackson', 'Liberty', 'Silver Republican', 'Socialist', 'Law and Order', 'Conservative', 'Federalist', 'Anti-Lecompton Democrat', 'Populist', 'Anti Masonic', 'Popular Democrat', 'Free Soil', 'Nonpartisan', 'Democrat-Liberal', 'Farmer-Labor', 'States Rights', 'Union Labor', 'Free Silver', 'Coalitionist', 'Constitutional Unionist', 'Ind. Republican', 'Unionist', 'American', 'Republican-Conservative', 'Progressive', 'Nullifier', 'Ind. Democrat', 'Pro-Administration', 'Ind. Republican-Democrat', 'Democratic Republican', 'Jackson', 'Crawford Republican', 'Liberal Republican', 'Readjuste

> gender and party columns contained an **empty string('')**. This indicates that one or more of the rows in the data have missing values in those columns. <br/>
Missing values are very common in real world data analysis, since the people compiling the datasets often don't have full information.<br/>
<br/>
성별 및 파티 열의 집합 표현에 빈 문자열 ( '')이 포함되어 있음을 알았을 수 있습니다. 이는 데이터의 하나 이상의 행에 해당 열에 누락 값이 있음을 나타냅니다. 누락 된 값은 실제 데이터 분석에서 매우 일반적입니다. 데이터 집합을 컴파일하는 사람들이 완전한 정보를 가지고 있지 않기 때문입니다.

<p style="color:Gray">Strategies to address missing data<p/><hr>
you can use one of the following strategies to address missing data.
* Remove any rows that contain missing data
* populate the empty fields with a specified value.
* populate the empty fields with a calculated value.
* Use analysis techniques that work with missing data

In [13]:
# focus on populating empty fields with a specified value
rows = [ ["Bassett", "Richard", "1745-04-02", "M", "sen", "DE", "Anti-Administration"],
    ["Bland", "Theodorick", "1742-03-21", "", "rep", "VA", ""]]
for row in rows:
    if row[6] == "":
        row[6] = "No Party"

In [14]:
print(header)

['last_name', 'first_name', 'birthday', 'gender', 'type', 'state', 'party']


In [15]:
# Replace any missing values in the gender column of legislators with the string "M"
for row in legislators:
    if row[3] == "":
        row[3] = "M"

In [16]:
gender = []
for item in legislators:
    gender.append(item[3])

In [17]:
print(set(gender))

{'M', 'F'}


In [18]:
print(legislators[:30])

[['Bassett', 'Richard', '1745-04-02', 'M', 'sen', 'DE', 'Anti-Administration'], ['Bland', 'Theodorick', '1742-03-21', 'M', 'rep', 'VA', ''], ['Burke', 'Aedanus', '1743-06-16', 'M', 'rep', 'SC', ''], ['Carroll', 'Daniel', '1730-07-22', 'M', 'rep', 'MD', ''], ['Clymer', 'George', '1739-03-16', 'M', 'rep', 'PA', ''], ['Contee', 'Benjamin', '', 'M', 'rep', 'MD', ''], ['Dalton', 'Tristram', '1738-05-28', 'M', 'sen', 'MA', 'Pro-Administration'], ['Elmer', 'Jonathan', '1745-11-29', 'M', 'sen', 'NJ', 'Pro-Administration'], ['Few', 'William', '1748-06-08', 'M', 'sen', 'GA', 'Anti-Administration'], ['Floyd', 'William', '1734-12-17', 'M', 'rep', 'NY', ''], ['Gale', 'George', '1756-06-03', 'M', 'rep', 'MD', ''], ['Grayson', 'William', '', 'M', 'sen', 'VA', 'Anti-Administration'], ['Grout', 'Jonathan', '1737-07-23', 'M', 'rep', 'MA', ''], ['Hawkins', 'Benjamin', '1754-08-15', 'M', 'sen', 'NC', 'Pro-Administration'], ['Huntington', 'Benjamin', '1736-04-19', 'M', 'rep', 'CT', ''], ['Izard', 'Ralph', 

<p style="color:Gray">Identify historical naming trends, and explore how popular names have changed form 1820 to today<p/><hr>
* looking for the most common names of U.S. legislators
* The year of birth

> The birthday column has the format 1820-01-02, which is hard to work with.<br/>
1. It's common to reformat values to simplify them.<br/>
2. split the data into its component parts

>생일 형식이 1820-01-02이면 작업하기 어렵습니다. 
1. 따라서 간단하게 값을 다시 형식화해야 합니다.
2. 날짜를 구성요소로 나눕니다.

In [19]:
print(header)

['last_name', 'first_name', 'birthday', 'gender', 'type', 'state', 'party']


In [20]:
date = "1820-01-02"
parts = date.split("-")
print(parts)

['1820', '01', '02']


In [21]:
birth_years = []
for item in legislators:
    parts = item[2].split('-')
    # print(parts[0])
    birth_years.append(parts[0])
print(birth_years[:10])

['1745', '1742', '1743', '1730', '1739', '', '1738', '1745', '1748', '1734']


* Converting a column to a different data type is a common operation in data analysis.

<p style="color:Gray">string form -> integers<p/><hr>
* date data
* To find the average year in which legislators born
* convert the data to integers
* int() function
> we just extracted the year on the previous screen, but it's in string form. To find the average year in which legislators were born, we'll need to convert the data to integers first. We can perform this conversion with the int() function. 
    1. handle the missing value - an empty string can't be converted to an integer.
    2. 

#### <p style="color:Tomato">Try/except<p/>
<p style="color:Gray">We can Manage errors with something known as a try/except block<p/><hr>

> Not all errors should halt execution, thogh. Sometimes we expect **a certain type of errors**, and want to handle it in a specific way that allows the code to complete.<br/>
If you surround the code that causes an error with a try/except block, the error will be handled, and the code will continue to run.

> 모든 오류가 실행을 중단해야하는 것은 아닙니다. 때로는 특정 유형의 오류가 예상되며 코드를 완료 할 수있는 특정 방식으로 처리하려고합니다.  try / except 블록에서 오류를 일으키는 코드를 둘러싸면 오류가 처리되고 코드가 계속 실행됩니다.

In [22]:
try:
    int('1')
except Exception:
    print("There was an error")

1

In [23]:
try:
    int('I am not a int')
except Exception:
    print("There was an error")

There was an error


In [24]:
string = float('hello')

ValueError: could not convert string to float: 'hello'

In [25]:
try:
    float("hello")
except Exception:
    print("Error converting to float.")

Error converting to float.


> try/eception
1. The python interpreter will try to run int(''), and generate a ValueError.
2. Instead of stopping the code from executing, it will be handled by the except statement, which will print the message "There was an error"
3. Python interpreter will countinue to run any lines of code that come after the cxcpt statment. 

> try/eception
1. 파이썬 인터프리터는 int()를 실행하고 ValueError를 나타내려고 한다.
2. 코드는 계속 실행되고, except문에 의해 처리된다. 오류 발생의 메세지를 출력한다.  
3. except문 다음에 나오는 코드를 계속 실행한다. 

<p style="color:Gray">Exception class<p/><hr>
When the Python interpreter generates an exception, it actually creates an instance of the Exception class.
* Has certain properties that help us **debug the error**
* Access the instance of the Exception class in the except statement body.<br/>
<br/>

파이썬 인터프리터가 예외를 생성 할 때 실제로 Exception 클래스의 인스턴스를 생성합니다.
* 오류를 디버그하는데 유용한 특정 속성을 가지고 있습니다. 
* except문의 본문에서 Exception 클래스의 인스턴스에 접근할 수 있습니다. 

In [26]:
try:
    int('')
except Exception as exc:
    print(type(exc)) # we can then access the vaiable in the except statement body.

<class 'ValueError'>


In [27]:
try:
    int('')
except Exception as exc:
    print(str(exc))

invalid literal for int() with base 10: ''


In [28]:
try:
    int('')
except Exception as exc:
    print(type(exc)) # Print the type of the Exception instance.
    print(str(exc)) # Convert the Exception instance to a string and print it out.

<class 'ValueError'>
invalid literal for int() with base 10: ''


In [29]:
numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
for i in numbers:
    try:
        int('')
    except Exception:
        print("There was an error")

There was an error
There was an error
There was an error
There was an error
There was an error
There was an error
There was an error
There was an error
There was an error
There was an error


<p style="color:Gray">Pass<p/><hr>
However, there are times when we don't want to do anything specific to handle eroors; we just want the code to keep running. <br/>
While the pass keyword doesn't actually do anything, it's a valid statement body. <br/>
<br/>
그러나 오류를 처리하기 위해 특정 작업을 수행하고 싶지 않은 경우가 있습니다. 코드를 계속 실행하기 만하면됩니다. pass 키워드는 실제로 아무 것도하지 않지만 유효한 문 본문입니다. 

In [30]:
numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
for i in numbers:
    try:
        int('')
    except Exception:
        pass

In [31]:
converted_years = []
for year in birth_years:
    try:
        year = int(year)
    except Exception:
        pass
    converted_years.append(year)

<p style="color:Gray">Convert all of the birth years in legislators to integers<p/><hr>
* To change the items in a list of lists.
* Need to loop over the top-level list(items)
* Convert all of the values in the second column of items to integers.
<br/>
<br/>
* 목록안의 목록의 항목을 변경합니다.
* 최상위 목록이 필요
* 항목의 두 번째 열에있는 모든 값을 정수로 변환

In [32]:
items = [          
        [1, "1", 2],    
        [2, "", 3],
        [5, "5", 3]
        ]

for item in items:
    item[1] = int(item[1])

ValueError: invalid literal for int() with base 10: ''

In [33]:
legi_dataset.header
checking_birthday = legi_dataset.column('birthday') # year-month-day

['last_name', 'first_name', 'birthday', 'gender', 'type', 'state', 'party']

In [34]:
for row in legislators:
    birthday = row[2]
    birth_year = birthday.split("-")[0] # birth_year = row[2].split("-")[0]
    try:
        birth_year = int(birth_year)
    except Exception:
        birth_year = 0
    row.append(birth_year)

In [35]:
print(legislators[:10])

[['Bassett', 'Richard', '1745-04-02', 'M', 'sen', 'DE', 'Anti-Administration', 1745], ['Bland', 'Theodorick', '1742-03-21', 'M', 'rep', 'VA', '', 1742], ['Burke', 'Aedanus', '1743-06-16', 'M', 'rep', 'SC', '', 1743], ['Carroll', 'Daniel', '1730-07-22', 'M', 'rep', 'MD', '', 1730], ['Clymer', 'George', '1739-03-16', 'M', 'rep', 'PA', '', 1739], ['Contee', 'Benjamin', '', 'M', 'rep', 'MD', '', 0], ['Dalton', 'Tristram', '1738-05-28', 'M', 'sen', 'MA', 'Pro-Administration', 1738], ['Elmer', 'Jonathan', '1745-11-29', 'M', 'sen', 'NJ', 'Pro-Administration', 1745], ['Few', 'William', '1748-06-08', 'M', 'sen', 'GA', 'Anti-Administration', 1748], ['Floyd', 'William', '1734-12-17', 'M', 'rep', 'NY', '', 1734]]


<p style="color:Gray">Using Chronological order situation<p/><hr>
>While exploring the dataset, you may have noticed that the legislators appear in roughly chronological order. 
<br/>

* The values generally **appear in chronological order**, we can loop through each year and replace any 0 values with the values from **the previous rows**
* By doing this, we'll make sure each legislator without a birth year is assigned one that's relatively close to the actual date.
<br/>
<br/>
* 값은 일반적으로 시간순을 표시되기 때문에 각 연도마다 loop를 돌리고 0값을 이전 행의 값으로 바꿀 수 있습니다. 
* 이렇게 함으로써 출생년도가 없는 의원에게 실제 날짜와 비교적 가깝게 값을 대입할 수 있습니다. 

1. We finished parsing the birth years to integers
2. we have several birth years with the value 0 
3. we replaced missing values with a fixed value M.
4. Using Chronological order Situation, to assign the value. 

<br/>

* 우리는 출생 년도를 정수로 파싱을 마쳤습니다.
* 0의 값을 몇 개 가지고 있습니다. 
* 누락된 값을 고정 값 M으로 바꾸었습니다. 
* 시간순으로 나타나는 데이터의 상황을 이용해서 0값에 적절한 값들을 대입시킬 수 있습니다. 

### <p style="color:Tomato">★<p/>

In [36]:
last_value = 1
for row in legislators:
    # print(row[7])
    # print("last_value:")
    # print(last_value)
    if row[7] == 0:
        row[7] = last_value
    last_value = row[7]

In [37]:
last_value

1963

In [38]:
legi_dataset.header

['last_name', 'first_name', 'birthday', 'gender', 'type', 'state', 'party']

[['Bassett', 'Richard', '1745-04-02', 'M', 'sen', 'DE', 'Anti-Administration', 1745], ['Bland', 'Theodorick', '1742-03-21', 'M', 'rep', 'VA', '', 1742], ['Burke', 'Aedanus', '1743-06-16', 'M', 'rep', 'SC', '', 1743], ['Carroll', 'Daniel', '1730-07-22', 'M', 'rep', 'MD', '', 1730], ['Clymer', 'George', '1739-03-16', 'M', 'rep', 'PA', '', 1739], ['Contee', 'Benjamin', '', 'M', 'rep', 'MD', '', <span style="color:Red">0</span>], ['Dalton', 'Tristram', '1738-05-28', 'M', 'sen', 'MA', 'Pro-Administration', 1738], ['Elmer', 'Jonathan', '1745-11-29', 'M', 'sen', 'NJ', 'Pro-Administration', 1745], ['Few', 'William', '1748-06-08', 'M', 'sen', 'GA', 'Anti-Administration', 1748], ['Floyd', 'William', '1734-12-17', 'M', 'rep', 'NY', '', 1734]]

In [39]:
print(legislators[:10])

[['Bassett', 'Richard', '1745-04-02', 'M', 'sen', 'DE', 'Anti-Administration', 1745], ['Bland', 'Theodorick', '1742-03-21', 'M', 'rep', 'VA', '', 1742], ['Burke', 'Aedanus', '1743-06-16', 'M', 'rep', 'SC', '', 1743], ['Carroll', 'Daniel', '1730-07-22', 'M', 'rep', 'MD', '', 1730], ['Clymer', 'George', '1739-03-16', 'M', 'rep', 'PA', '', 1739], ['Contee', 'Benjamin', '', 'M', 'rep', 'MD', '', 1739], ['Dalton', 'Tristram', '1738-05-28', 'M', 'sen', 'MA', 'Pro-Administration', 1738], ['Elmer', 'Jonathan', '1745-11-29', 'M', 'sen', 'NJ', 'Pro-Administration', 1745], ['Few', 'William', '1748-06-08', 'M', 'sen', 'GA', 'Anti-Administration', 1748], ['Floyd', 'William', '1734-12-17', 'M', 'rep', 'NY', '', 1734]]


In [40]:
for row in legislators:
    if row[6] == "":
        row[6] = "No Party"

In [41]:
print(legislators[:10])

[['Bassett', 'Richard', '1745-04-02', 'M', 'sen', 'DE', 'Anti-Administration', 1745], ['Bland', 'Theodorick', '1742-03-21', 'M', 'rep', 'VA', 'No Party', 1742], ['Burke', 'Aedanus', '1743-06-16', 'M', 'rep', 'SC', 'No Party', 1743], ['Carroll', 'Daniel', '1730-07-22', 'M', 'rep', 'MD', 'No Party', 1730], ['Clymer', 'George', '1739-03-16', 'M', 'rep', 'PA', 'No Party', 1739], ['Contee', 'Benjamin', '', 'M', 'rep', 'MD', 'No Party', 1739], ['Dalton', 'Tristram', '1738-05-28', 'M', 'sen', 'MA', 'Pro-Administration', 1738], ['Elmer', 'Jonathan', '1745-11-29', 'M', 'sen', 'NJ', 'Pro-Administration', 1745], ['Few', 'William', '1748-06-08', 'M', 'sen', 'GA', 'Anti-Administration', 1748], ['Floyd', 'William', '1734-12-17', 'M', 'rep', 'NY', 'No Party', 1734]]


# # List Comprehensions
### <p style="color:Tomato">Learn about list comprehensions and the None type while finding common names for U.S. legislators.<p/>

In [42]:
animals = ["Dog", "Tiger", "SuperLion", "Cow", "Panda"]
viciousness = [1, 5, 10, 10, 1]
for animal in animals:
    print("Animal")
    print(animal)
    print("Viciousness")

Animal
Dog
Viciousness
Animal
Tiger
Viciousness
Animal
SuperLion
Viciousness
Animal
Cow
Viciousness
Animal
Panda
Viciousness


In [43]:
for i, animal in enumerate(animals):
    print("Animal Index")
    print(i)
    print("Animal")
    print(animal)

Animal Index
0
Animal
Dog
Animal Index
1
Animal
Tiger
Animal Index
2
Animal
SuperLion
Animal Index
3
Animal
Cow
Animal Index
4
Animal
Panda


In [44]:
animals = ["Dog", "Tiger", "SuperLion", "Cow", "Panda"]
viciousness = [1, 5, 10, 10, 1]

for i, animal in enumerate(animals):
    print("Animal")
    print(animal)
    print("Viciousness")
    print(viciousness[i])

Animal
Dog
Viciousness
1
Animal
Tiger
Viciousness
5
Animal
SuperLion
Viciousness
10
Animal
Cow
Viciousness
10
Animal
Panda
Viciousness
1
